autoprecompiles/src/lib.rs

```diff
@@ -510,11 +524,34 @@ pub fn exec_receive<T: FieldElement>(machine: &SymbolicMachine<T>) -> SymbolicBu
     r
 }
 
+// Check that the number of register memory bus interactions in the precompile is even.
+// The check is performed on the total number and per address.
```

Comment:
@Schaeff here

Reply:
```rust
// Check that the number of register memory bus interactions for each concrete address in the precompile is even.
// Assumption: all register memory bus interactions feature a concrete address
```

then we don't have to count the total?

But maybe I'm misunderstanding something

Reply:
I think it's safer, but sure can remove